### PR TITLE
chore: add devcontainer.json

### PR DESCRIPTION
**Pull request by ylemkimon**
Configuration for GitHub Codespaces and Visual Studio Codespaces.

### CODE REVIEW

File: webpack.dev.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ katexConfig.devServer = {
     disableHostCheck: true,
     host: '0.0.0.0',
     port: PORT,
+    sockPort: 'location',
```
Comment (kevinbarabash) [inline webpack.dev.js:22]: The docs do say anything about `"location"`.  What does this setting do?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "KaTeX Codespaces",
+  "image": "mcr.microsoft.com/vscode/devcontainers/javascript-node:14",
+  "extensions": [
+    "dbaeumer.vscode-eslint",
+    "flowtype.flow-for-vscode",
+    "stylelint.vscode-stylelint",
+    "arcanis.vscode-zipfs"
```
Comment (kevinbarabash) [inline .devcontainer/devcontainer.json:8]: What is this used for?

**Review by kevinbarabash — COMMENTED** [review]
I'm excited to give this a try.  Once this PR is merged with the "Open codespace" button just appear in GitHub?

File: webpack.dev.js (hunk 1 above)
Comment (ylemkimon) [reply webpack.dev.js:22]: @kevinbarabash It makes `webpack-dev-server` to use the browser's location port as the socket port: https://github.com/webpack/webpack-dev-server/pull/2341. For instance, when developing locally in `http://localhost:7936`, it uses `7936`, and using GitPod or codespaces, they are tunneled using HTTPS, so it uses `443`. This eliminates the need to set `--sockPort 443` manually.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (ylemkimon) [reply .devcontainer/devcontainer.json:8]: @kevinbarabash Yarn 2 uses a zip-based file system, so this extension allows us to browse into a zip file in vscode.

**Comment by ylemkimon** [comment]
Since it's still in limited beta, it can be started from https://github.com/codespaces. Maybe when it's general availability, a badge or button to start it may be added.

**Review by kevinbarabash — APPROVED** [review]
Thanks for adding support for Codespaces.

**Pull request merged**